### PR TITLE
Issue #2335: [UX] Project Browser: Add links with helper actions (con...

### LINKED ISSUE #2335: [UX] Project Browser: Add a link to the project config page if already installed.
If a project is already downloaded and installed, there is a "already enabled" indicator in the place of the "add to installation queue" link:

backdrop-project_browser-project_already_enabled

How about we turned this to a link to the project config page (if the project has one)?

- for already installed layouts, this link could be static, pointing to `/admin/structure/layouts/settings`
- for already installed themes, we should be linking to `/appearance/settings/%theme_name%` (if the teme has settings)
- for already installed modules, we should be linking to whatever is in the `configure =` entry in their .info files

### PR DESCRIPTION
**Pull request by klonos**
…figure/settings/enable) next to project installation status.

Fixes https://github.com/backdrop/backdrop-issues/issues/2335

Replaces https://github.com/backdrop/backdrop/pull/1646

Depends on https://github.com/backdrop/backdrop/pull/1644

### CODE REVIEW

File: core/modules/installer/installer.theme.inc [hunk 1]
```diff
@@ -175,18 +175,87 @@ function template_preprocess_installer_browser_project(&$variables) {
...
+        break;
+    }
+
+    $status = '<div class="install-disabled">' . t('Already enabled');
+    // If the project has a config page (for modules), or a settings page (for
+    // themes, or is a layout template (that can always be enabled/disabled),...
+    if (isset($action) && isset($url)) {
+      // ...then render the respective action link after the "Already enabled"
+      // status...
+      $status .= ', ' . t('<a href="@url">@action</a>', array('@url' => url($url), '@action' => $action));
```
Comment (quicksketch) [inline core/modules/installer/installer.theme.inc]: This t() call is not going to be useful to translators. We should instead add `t()` around the strings themselves.

File: core/modules/installer/installer.theme.inc [hunk 2]
```diff
@@ -175,18 +175,87 @@ function template_preprocess_installer_browser_project(&$variables) {
...
+        // If the theme has settings.
+        if (backdrop_theme_access($project['name']) && theme_has_settings($project['name'])) {
+          $action = 'settings';
+          $url = 'admin/appearance/settings/' . $project['name'];
+        }
+        break;
+      case 'layout':
+        // All layout themplates can be enabled/disabled in the "Layout
+        // settings" page.
+        $action = 'settings';
```
Comment (quicksketch) [inline core/modules/installer/installer.theme.inc]: Wrap `settings` in `t()` here.

File: core/modules/installer/installer.theme.inc [hunk 3]
```diff
@@ -175,18 +175,87 @@ function template_preprocess_installer_browser_project(&$variables) {
...
+        // If the module has a coniguration page defined in its .info file.
+        if (isset($project_info->info['configure'])) {
+          $action = 'configure';
+          $url = $project_info->info['configure'];
+        }
+        break;
+      case 'theme':
+        // If the theme has settings.
+        if (backdrop_theme_access($project['name']) && theme_has_settings($project['name'])) {
+          $action = 'settings';
```
Comment (quicksketch) [inline core/modules/installer/installer.theme.inc]: Wrap `settings` in `t()` here.

File: core/modules/installer/installer.theme.inc [hunk 4]
```diff
@@ -175,18 +175,87 @@ function template_preprocess_installer_browser_project(&$variables) {
...
-    $variables['install'] = '';
+
+    // ...set the action text and link based on project type.
+    switch ($project['type']) {
+      case 'module':
+        $all_modules_info = system_rebuild_module_data();
+        $project_info = $all_modules_info[$project['name']];
+        // If the module has a coniguration page defined in its .info file.
+        if (isset($project_info->info['configure'])) {
+          $action = 'configure';
```
Comment (quicksketch) [inline core/modules/installer/installer.theme.inc]: Wrap `configure` in `t()` here.

File: core/modules/installer/installer.theme.inc [hunk 5]
```diff
@@ -175,18 +175,87 @@ function template_preprocess_installer_browser_project(&$variables) {
...
+        break;
+      case 'layout':
+        // All layout themplates can be enabled/disabled in the "Layout
+        // settings" page.
+        $action = 'settings';
+        $url = 'admin/structure/layouts/settings';
+        break;
+    }
+
+    $status = '<div class="install-disabled">' . t('Already enabled');
```
Comment (quicksketch) [inline core/modules/installer/installer.theme.inc]: Minor, but I think a span may be more appropriate here than a div?

File: core/modules/installer/installer.theme.inc [hunk 6]
```diff
@@ -175,18 +175,87 @@ function template_preprocess_installer_browser_project(&$variables) {
...
+        break;
+      case 'theme':
+        $url = 'admin/appearance/list';
+        break;
+      case 'layout':
+        $url = 'admin/structure/layouts/settings';
+        break;
+    }
+    // ...render the status text...
+    $status = '<div class="install-disabled">' . t('Already downloaded');
```
Comment (quicksketch) [inline core/modules/installer/installer.theme.inc]: Again div -> span

**Review by quicksketch — CHANGES_REQUESTED** [review]
Looking good! But some minor issues regarding t() that need to be addressed.

File: core/modules/installer/installer.theme.inc (hunk 2 above)
Comment (klonos) [reply core/modules/installer/installer.theme.inc]: Done.

File: core/modules/installer/installer.theme.inc (hunk 3 above)
Comment (klonos) [reply core/modules/installer/installer.theme.inc]: Done.

File: core/modules/installer/installer.theme.inc (hunk 4 above)
Comment (klonos) [reply core/modules/installer/installer.theme.inc]: Done.

File: core/modules/installer/installer.theme.inc (hunk 5 above)
Comment (klonos) [reply core/modules/installer/installer.theme.inc]: Done.

File: core/modules/installer/installer.theme.inc (hunk 6 above)
Comment (klonos) [reply core/modules/installer/installer.theme.inc]: Done.

File: core/modules/installer/installer.theme.inc (hunk 1 above)
Comment (klonos) [reply core/modules/installer/installer.theme.inc]: Fixed.

**Comment by klonos** [comment]
...lets see how badly this may break after refreshing the PR sandbox 😅

**Comment by klonos** [comment]
...replaced by https://github.com/backdrop/backdrop/pull/2797

**Pull request closed without merging**